### PR TITLE
Added nexus_upload action

### PR DESCRIPTION
**Pull request by xfreebird**
Action that can upload any artefact to Nexus server.

The failing tests are not related to this PR.

### CODE REVIEW

File: lib/fastlane/actions/nexus_upload.rb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        options << "-F e=#{file_extension}"
+        options << "-F file=@#{file_path}"
+        options << "-u #{params[:username].shellescape}:#{params[:password].shellescape}"
+
+        options
+      end
+
+      def self.ssl_options(params)
+        options = []
+        unless params[:ssl_verify]
```
Comment (KrauseFx) [inline lib/fastlane/actions/nexus_upload.rb]: Why would we want to have this option?

File: lib/fastlane/actions/nexus_upload.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                                       optional: true)
+        ]
+      end
+
+      def self.authors
+        ["xfreebird"]
+      end
+
+      def self.is_supported?(platform)
+        true
```
Comment (KrauseFx) [inline lib/fastlane/actions/nexus_upload.rb]: Is this only for iOS?

File: lib/fastlane/actions/nexus_upload.rb [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :proxy_address,
+                                       env_name: "FL_NEXUS_PROXY_ADDRESS",
+                                       description: "Proxy address",
+                                       default_value: nil,
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :proxy_port,
+                                       env_name: "FL_NEXUS_PROXY_PORT",
+                                       description: "Proxy port",
+                                       default_value: nil,
```
Comment (KrauseFx) [inline lib/fastlane/actions/nexus_upload.rb]: You can remove the `default_value: nil`

**Comment by KrauseFx** [comment]
Thanks - this looks good, I added some comments before it's ready to merge :+1:

File: lib/fastlane/actions/nexus_upload.rb (hunk 1 above)
Comment (xfreebird) [inline lib/fastlane/actions/nexus_upload.rb]: In some organisations Nexus will be hosted on internal servers, with self signed SSL/TLS certificate. And to be able to upload it is needed to disable ssl check.

File: lib/fastlane/actions/nexus_upload.rb (hunk 2 above)
Comment (xfreebird) [inline lib/fastlane/actions/nexus_upload.rb]: It is universal, it can store any kind of file.

File: lib/fastlane/actions/nexus_upload.rb (hunk 1 above)
Comment (KrauseFx) [inline lib/fastlane/actions/nexus_upload.rb]: Makes sense, thanks :+1:

**Comment by KrauseFx** [comment]
Thanks, I'll take over now. Thanks again for your contribution :+1:

**Comment by xfreebird** [comment]
Thanks for the help :+1:

**Pull request merged**